rust/operator-binary/src/controller.rs

```diff
@@ -988,15 +1102,39 @@ fn build_opa_start_command(merged_config: &OpaConfig, container_name: &str) -> S
         create_vector_shutdown_file_command =
             create_vector_shutdown_file_command(STACKABLE_LOG_DIR),
         shutdown_grace_period_s = merged_config.graceful_shutdown_timeout.unwrap_or(DEFAULT_SERVER_GRACEFUL_SHUTDOWN_TIMEOUT).as_secs(),
-        // Redirects matter!
-        // We need to watch out, that the following "$!" call returns the PID of the main (opa-bundle-builder) process,
-        // and not some utility (e.g. multilog or tee) process.
-        // See https://stackoverflow.com/a/8048493
-        logging_redirects = if console_logging_off {
-            format!(" &> >(/stackable/multilog s{OPA_ROLLING_LOG_FILE_SIZE_BYTES} n{OPA_ROLLING_LOG_FILES} {STACKABLE_LOG_DIR}/{container_name})")
-        } else {
-            format!(" &> >(tee >(/stackable/multilog s{OPA_ROLLING_LOG_FILE_SIZE_BYTES} n{OPA_ROLLING_LOG_FILES} {STACKABLE_LOG_DIR}/{container_name}))")
+        opa_log_level = get_most_expressive_log_level(&[console_log_level, file_log_level]).unwrap_or(LogLevel::INFO).to_opa_literal()
+    }
+}
+
+fn get_most_expressive_log_level(log_levels: &[LogLevel]) -> Option<LogLevel> {
+    let mut log_levels_mapped: Vec<u8> = Vec::new();
+
+    for log_level in log_levels.iter() {
+        match log_level {
+            LogLevel::TRACE => log_levels_mapped.push(6),
+            LogLevel::DEBUG => log_levels_mapped.push(5),
+            LogLevel::INFO => log_levels_mapped.push(4),
+            LogLevel::WARN => log_levels_mapped.push(3),
+            LogLevel::ERROR => log_levels_mapped.push(2),
+            LogLevel::FATAL => log_levels_mapped.push(1),
+            LogLevel::NONE => log_levels_mapped.push(0),
+        }
+    }
+
+    let max_value = log_levels_mapped.iter().max();
+
+    match max_value {
+        Some(max) => match max {
+            6 => Some(LogLevel::TRACE),
+            5 => Some(LogLevel::DEBUG),
+            4 => Some(LogLevel::INFO),
+            3 => Some(LogLevel::WARN),
+            2 => Some(LogLevel::ERROR),
+            1 => Some(LogLevel::FATAL),
+            0 => Some(LogLevel::NONE),
+            _ => None,
         },
+        None => None,
```

Comment:
note: This can be replaced by `log_levels.iter.min()` because `LogLevel` derives `PartialOrd`.

Reply:
Also see: https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html#derivable

Reply:
Oh that's awesome! I was actually looking for that but missed the derive above the derivative for some reason. Darn it, could've saved me some time :see_no_evil: thank you!

Reply:
Sure thing :)